integration_tests/testsuite/deploy_app.py

```diff
@@ -38,21 +38,39 @@ def _set_app_image(image):
     fin.close()
 
 
-def deploy_app(image, appdir):
+def _set_builder_image(builder):
+    with open('test.yaml.in', 'r') as fin:
+        with open('test.yaml', 'w') as fout:
+            for line in fin:
+                fout.write(line.replace('${STAGING_BUILDER_IMAGE}', builder))
+        fout.close()
+    fin.close()
+
+
+def deploy_app(base_image, builder_image, appdir, yaml):
     try:
+        if yaml:
+            # convert yaml to absolute path before changing directory
+            yaml = os.path.abspath(yaml)
+
         # change to app directory (and remember original directory)
         owd = os.getcwd()
         os.chdir(appdir)
 
-        # fills in image field in templated Dockerfile if image is specified
-        if image:
-            _set_app_image(image)
+        # fills in image field in templated Dockerfile and/or builder yaml
+        if base_image:
+            _set_base_image(base_image)
+        if builder_image:
+            _set_builder_image(builder_image)
 
         deployed_version = test_util.generate_version()
 
         # TODO: once sdk driver is published, use it here
-        deploy_command = ['gcloud', 'app', 'deploy', '--no-promote'
+        deploy_command = ['gcloud', 'app', 'deploy', '--no-promote',
                           '--version', deployed_version, '-q']
+        if yaml:
+            logging.info(yaml)
+            deploy_command.append(yaml)
```

Comment:
positional, it's the last argument. if it isn't provided, gcloud assumes it's `<current_directory>/app.yaml`.